lua/ide/components/explorer/component.lua

```diff
@@ -272,9 +272,11 @@ ExplorerComponent.new = function(name, config)
             fnode.touch(input)
             self.tree.marshal()
             self.state["cursor"].restore()
-            if self.config.edit_on_create then
+            local path = fnode.path .. "/" .. input
+            -- only edit if the created path is a file, not a directory
+            if self.config.edit_on_create and not vim.endswith(input, '/') then
```

Comment:
filenodes have a .kind field right? I believe just check this for "dir" str

Reply:
See example in the marshal function

Reply:
Yeah but I think in this case the filenode is for the containing directory, the newly created directory/file hasn't had a filenode created for it yet.